modules/terraform/output.go

```diff
@@ -49,6 +50,131 @@ func OutputRequiredE(t testing.TestingT, options *Options, key string) (string,
 	return out, nil
 }
 
+func parseListOfMaps(l []interface{}) ([]map[string]interface{}, error) {
+	var result []map[string]interface{}
+
+	for _, v := range l {
+		m, err := parseMap(v.(map[string]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, m)
+	}
+
+	return result, nil
+
+}
+
+// parseMap takes a map of interfaces and parses the types. Is also recursive for nested objects.
+func parseMap(m map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
I'm a bit confused... If this is already a `map[string]interface{}` and you're returning a `map[string]interface{}`, what is changing?

Reply:
I think you're asking why I have a second `parseMap` function? If so, I have it to support recursive functions. See https://github.com/gruntwork-io/terratest/pull/513/files#diff-40b865b3d1190588582bbeaba6461217R76-R81

Reply:
No, what I meant was this: you have this `parseMap` function that takes as input somemthing that is _already_ a `map[string]interface{}`. So what parsing, exactly, are you doing? 

I can see that within `parseMap`, you loop over the keys and values of the passed-in `map`, look at the type of each value, and if the value is:

- Already a map or list, you call `parseMap` recursively... But the type stays the same.
- Boolean: the type stays the same.
- Float: the type stays the same, unless it's an int.
- Everything else: you convert it to a string.

My question is... Do we need to do this "parsing" at all? Or could we just leave it as-is, based on what you get back from `json.Unmarshall`?

Reply:
Ahh so sorry. I completely misunderstood the question. So the reason I had to go through all this is due to how [Unmarshal()](https://golang.org/pkg/encoding/json/#Unmarshal) converts all numbers to `float`s, including `int`s.

This was causing my tests to fail and I figured it wouldn't be intuitive for developers to write their tests using floats instead of ints. Although, thanks to your review I do see that I needlessly set up case statements for `bool` and `float`. I will update my PR to reflect this.

Sorry again for the misunderstanding, I'm still pretty new to go.

FYI, this is what the test looks like when I don't do the number parsing:

```
TestOutputMapOfObjects 2020-04-30T10:02:42-04:00 command.go:158: {"nest":{"five":"five","four":4},"nest_list":[{"seven":"seven","six":6}],"one":1,"somebool":true,"somefloat":1.1,"three":"three","two":"two"}
    TestOutputMapOfObjects: output_test.go:130:
                Error Trace:    output_test.go:130
                Error:          Not equal:
                                expected: map[string]interface {}{"nest":map[string]interface {}{"five":"five", "four":4}, "nest_list":[]map[string]interface {}{map[string]interface {}{"seven":"seven", "six":6}}, "one":1, "somebool":true, "somefloat":1.1, "three":"three", "two":"two"}
                                actual  : map[string]interface {}{"nest":map[string]interface {}{"five":"five", "four":4}, "nest_list":[]map[string]interface {}{map[string]interface {}{"seven":"seven", "six":6}}, "one":1, "somebool":true, "somefloat":1.1, "three":"three", "two":"two"}

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,3 +3,3 @@
                                   (string) (len=4) "five": (string) (len=4) "five",
                                -  (string) (len=4) "four": (int) 4
                                +  (string) (len=4) "four": (float64) 4
                                  },
                                @@ -8,6 +8,6 @@
                                    (string) (len=5) "seven": (string) (len=5) "seven",
                                -   (string) (len=3) "six": (int) 6
                                +   (string) (len=3) "six": (float64) 6
                                   }
                                  },
                                - (string) (len=3) "one": (int) 1,
                                + (string) (len=3) "one": (float64) 1,
                                  (string) (len=8) "somebool": (bool) true,
                Test:           TestOutputMapOfObjects
```

Reply:
I will definitely add comments to the new functions, thanks! I actually had an iteration that used `useNumber` along with `Decoder`. But this resulted in more code and the same end result we have here (in terms of how `int`s are handled).

In my mind, if somebody passes:

```
{
  "foo": 1,
  "bar": 1.3
}
```
I would expect `foo` to always return an int. However, with my changes, if they instead had:
```
{
  "foo": 1.0,
  "bar": 1.3
}
```
Then `foo` will still be an int (because of using ParseInt). I think this could be resolved by documentation stating a test will need to manually convert from the `int` to a `float` if that functionality is needed. Think that would be ok?